### PR TITLE
Fixed wrong type assertion for received transactions

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Fixed wrong type assertion for received transactions
* Added guard checks for message type and message round in consensus topic interceptor
* Fixed memory leak in miniblocks/cross txs pools cleaner when pending miniblocks would not reached to 0 for a long time
* Added guard check for received header hash size from consensus message in consensus topic interceptor

### CODE REVIEW

File: process/block/poolsCleaner/crossTxsPoolsCleaner.go [hunk 1]
```diff
@@ -115,7 +128,19 @@ func (ctpc *crossTxsPoolsCleaner) receivedRewardTx(key []byte, value interface{}
...
+
+	tx, ok := value.(data.TransactionHandler)
+	if !ok {
+		log.Warn("crossTxsPoolsCleaner.receivedRewardTx", "error", process.ErrWrongTypeAssertion)
+		return
+	}
+
+	senderShardID, receiverShardID, err := ctpc.computeSenderAndReceiverShards(tx)
+	if err != nil {
+		log.Debug("crossTxsPoolsCleaner.receivedRewardTx", "error", err.Error())
```
Comment (AdoAdoAdo) [inline process/block/poolsCleaner/crossTxsPoolsCleaner.go:140]: shouldn't you return here on error as well?

File: process/block/poolsCleaner/crossTxsPoolsCleaner.go [hunk 2]
```diff
@@ -124,17 +149,28 @@ func (ctpc *crossTxsPoolsCleaner) receivedUnsignedTx(key []byte, value interface
...
 	if !ok {
-		log.Warn("crossTxsPoolsCleaner.processReceivedTx", "error", process.ErrWrongTypeAssertion)
+		log.Warn("crossTxsPoolsCleaner.receivedUnsignedTx", "error", process.ErrWrongTypeAssertion)
 		return
 	}
 
-	if wrappedTx.SenderShardID == ctpc.shardCoordinator.SelfId() {
+	senderShardID, receiverShardID, err := ctpc.computeSenderAndReceiverShards(tx)
+	if err != nil {
+		log.Debug("crossTxsPoolsCleaner.receivedUnsignedTx", "error", err.Error())
```
Comment (AdoAdoAdo) [inline process/block/poolsCleaner/crossTxsPoolsCleaner.go:161]: return on error?

File: process/block/poolsCleaner/crossTxsPoolsCleaner.go (hunk 1 above)
Comment (SebastianMarian) [reply process/block/poolsCleaner/crossTxsPoolsCleaner.go:140]: Yep, forgot to do it

File: process/block/poolsCleaner/crossTxsPoolsCleaner.go (hunk 2 above)
Comment (SebastianMarian) [reply process/block/poolsCleaner/crossTxsPoolsCleaner.go:162]: Yep, forgot to do it

**Pull request merged**